File: src/main/java/com/geekday/RestServer.java
package com.geekday;

import com.geekday.accounting.account.domain.AccountRepository;
import com.geekday.accounting.account.domain.CustomerSubscriber;
import com.geekday.accounting.customer.domain.AccountSubscriber;
import com.geekday.accounting.customer.domain.CustomerRepository;
import com.geekday.accounting.account.web.AccountResource;
import com.geekday.accounting.customer.web.CustomerResource;
import com.geekday.common.DomainEventPublisher;
import org.eclipse.jetty.server.Server;
import org.eclipse.jetty.servlet.ServletContextHandler;
import org.eclipse.jetty.servlet.ServletHolder;


public class RestServer {

    public static void main(String[] args) throws Exception {
        initializeApplication();
        startWebServer();
    }

    private static void initializeApplication() {
        CustomerRepository.initialize();
        AccountRepository.initialize();

        DomainEventPublisher.getInstance();
        new CustomerSubscriber().start();
        new AccountSubscriber().start();
    }

    private static void startWebServer() {
        ServletContextHandler context = new ServletContextHandler(ServletContextHandler.SESSIONS);
        context.setContextPath("/");

        Server jettyServer = new Server(8080);
        jettyServer.setHandler(context);

        ServletHolder jerseyServlet = context.addServlet(
                org.glassfish.jersey.servlet.ServletContainer.class, "/*");
        jerseyServlet.setInitOrder(0);

        jerseyServlet.setInitParameter(
                "jersey.config.server.provider.classnames",
                CustomerResource.class.getCanonicalName()+"," +
                        AccountResource.class.getCanonicalName());

        try {
            System.out.println("jerseyServlet = " + jerseyServlet);
            jettyServer.start();
            jettyServer.join();
        } catch(Throwable t) {
            t.printStackTrace();
        }finally {
            jettyServer.destroy();
        }
    }
}

File: src/main/scala/com/geekday/RestServer.scala
package com.geekday

import akka.actor.{ActorSystem, Props}
import com.geekday.accounting.AccountActor
import com.geekday.accounting.account.domain.{AccountRepository, CustomerSubscriber}
import com.geekday.accounting.account.web.AccountResource
import com.geekday.accounting.customer.domain.{AccountSubscriber, CustomerRepository}
import com.geekday.accounting.customer.web.CustomerResource
import com.geekday.common.DomainEventPublisher
import org.eclipse.jetty.server.Server
import org.eclipse.jetty.servlet.ServletContextHandler
import org.glassfish.jersey.servlet.ServletContainer

object RestServer {
  @throws[Exception]
  def main(args: Array[String]): Unit = {
    val system = ActorSystem("micro-actor")
    val actorRef = system.actorOf(Props(classOf[AccountActor]))
    actorRef ! "update"

    initializeApplication()
    startWebServer()
  }

  private def initializeApplication() = {
    CustomerRepository.initialize()
    AccountRepository.initialize()
    DomainEventPublisher.start
    new CustomerSubscriber().start()
    new AccountSubscriber().start()
  }

  private def startWebServer() = {
    val context = new ServletContextHandler(ServletContextHandler.SESSIONS)
    context.setContextPath("/")
    val jettyServer = new Server(8051)
    jettyServer.setHandler(context)
    val jerseyServlet = context.addServlet(classOf[ServletContainer], "/*")
    jerseyServlet.setInitOrder(0)
    jerseyServlet.setInitParameter("jersey.config.server.provider.classnames", classOf[CustomerResource].getCanonicalName + "," + classOf[AccountResource].getCanonicalName)
    try {
      System.out.println("jerseyServlet = " + jerseyServlet)
      jettyServer.start()
      jettyServer.join()
    } catch {
      case t: Throwable ⇒
        t.printStackTrace()
    } finally jettyServer.destroy()
  }
}